pkg/bff/models/v1/registration.go

```diff
@@ -130,9 +132,54 @@ func (r *RegistrationForm) ValidateLegalPerson() ValidationErrors {
 }
 
 // Validate only the fields in the contacts step.
-func (r *RegistrationForm) ValidateContacts() ValidationErrors {
-	// TODO: implement
-	return nil
+func (r *RegistrationForm) ValidateContacts() (v ValidationErrors) {
+	if r.Contacts == nil {
+		v = append(v, &ValidationError{Field: "contacts", Err: ErrMissingField.Error()})
+		return v
+	}
+
+	// ValidateContact handles the case where the contact is nil
+	v, _ = v.Append(ValidateContact(r.Contacts.Technical, "contacts.technical"))
+	v, _ = v.Append(ValidateContact(r.Contacts.Administrative, "contacts.administrative"))
+	v, _ = v.Append(ValidateContact(r.Contacts.Legal, "contacts.legal"))
+	v, _ = v.Append(ValidateContact(r.Contacts.Billing, "contacts.billing"))
+
+	return v
+}
+
+// Validate a single contact, using the field name to construct errors.
+func ValidateContact(contact *pb.Contact, fieldName string) (v ValidationErrors) {
+	if contact == nil {
+		v = append(v, &ValidationError{Field: fieldName, Err: ErrMissingField.Error()})
+		return v
+	}
```

Comment:
I don't remember, are we requiring each contact to be filled in on the BFF form on the frontend?

Reply:
Ok, the rules are:

1. At least one contact is specified (and if only one is specified it must be the administrative contact) 
2. Either administrative or technical contact is required
3. Either administrative or legal contact is required

So the billing contact is completely optional; only administrative is valid, as is technical + legal.

Kind of a mess I know.

Reply:
Ok, how are we communicating this to the user? Should we return an error like "administrative or technical is required" if they only filled in the legal contact?